plugins/flatpak/gs-flatpak.c

```diff
@@ -2577,6 +2577,43 @@ gs_flatpak_refine_appstream (GsFlatpak *self, GsApp *app, GError **error)
 	return TRUE;
 }
 
+static void
+gs_flatpak_refine_proxy_app (GsFlatpak	*self,
+			     GsApp	*app)
+{
+	GPtrArray *proxied_apps = gs_app_get_related (app);
+	g_autoptr(GsAppList) real_related_apps = NULL;
+
+	if (proxied_apps->len == 0)
+		return;
+
+	real_related_apps = gs_app_list_new ();
+
+	/* get a list of the real apps (not eventual wildcard ones) */
+	for (guint i = 0; i < proxied_apps->len; ++i) {
+		GsApp *proxied_app = g_ptr_array_index (proxied_apps, i);
+		GsApp *app_cached = proxied_app;
+
+		if (gs_app_has_quirk (proxied_app, GS_APP_QUIRK_IS_WILDCARD) &&
+		    gs_app_get_bundle_kind (proxied_app) == AS_BUNDLE_KIND_FLATPAK) {
+			const gchar *id = gs_app_get_unique_id (proxied_app);
+			app_cached = gs_plugin_cache_lookup (self->plugin, id);
+			if (app_cached == NULL)
+				continue;
+		}
+
+		gs_app_list_add (real_related_apps, app_cached);
+	}
+
+	/* set the cached apps as the related ones instead */
+	gs_app_clear_related (app);
+
+	for (guint i = 0; i < gs_app_list_length (real_related_apps); ++i) {
```

Comment:
Looks like this loop could be folded into the one above.

Reply:
Hmm, we're clearing the related apps of the app in the call above (meaning it does clear the object it returns). So for me to fold this into the loop above I'd have to copy those apps elsewhere.

Reply:
Or move the `gs_app_clear_related (app)` call above the first loop?

Reply:
That means we wouldn't be able to loop through the `GPtrArray` we assigned before because it'll be cleared in the `gs_app_clear_related` call.